### ISSUE TITLE
ONNX model outputs an array of null

### ISSUE THREAD

**Issue by sidharthg-couture** [kibae/onnxruntime-server#90]
I have configured my ONNX model (based on stella_400M) to have a fixed batch size for inputs and outputs. It even runs with response 200 on the server. 

My issue is the entire output of the result is an array of `null` of size 1024 (which is expected, that is the dimension of the model output). When I use the same ONNX model with the python ONNXRuntime library, it works as expected.

- Output from Python ONNX-Runtime 
```array([ 0.10638401,  0.09311324, -0.555768  , ...,  0.99843943,
        0.2404    ,  0.164792  ], shape=(1024,), dtype=float32)```

- Output from ONNX-Runtime on onnxruntime-server
```Response Body: {'sentence_embedding': [[None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, ... None, None]```

Would you have any idea why this would be happening? Or any possible fixes?
Thanks!

**Comment by kibae** [comment]
Hello. @sidharthg-couture  :)

Could you please display the ONNX session information? It can be queried via HTTP GET `/api/sessions` or HTTP GET `/api/sessions/{model}/{version}`. It would be great if you could show the response from this API.

**Comment by sidharthg-couture** [comment]
Hi, here is the response from the API:
```[
  {
    "created_at": 1738662696,
    "execution_count": 0,
    "inputs": {
      "attention_mask": "int64[1,-1]",
      "input_ids": "int64[1,-1]",
      "token_type_ids": "int64[1,-1]"
    },
    "last_executed_at": 0,
    "model": "stella-v3",
    "option": {
      "cuda": false
    },
    "outputs": {
      "sentence_embedding": "float32[-1,1024]"
    },
    "version": "model"
  }
]```

**Comment by kibae** [comment]
Hello, @sidharthg-couture
Thank you for providing the information. :)

According to the response from the Python code, the shape appears to be `(1024, -1)`.
- Output from Python ONNX-Runtime
```
array([ 0.10638401,  0.09311324, -0.555768  , ...,  0.99843943,
        0.2404    ,  0.164792  ], shape=(1024,), dtype=float32)
```

However, the response from the ONNX session API indicates that the shape of `sentence_embedding` is `(-1, 1024)`.
```
    "outputs": {
      "sentence_embedding": "float32[-1,1024]"
    },
```
How about modifying the `dynamic_axes` values during the ONNX export so that the shape becomes `(-1, 1024)`?
```
    dynamic_axes={
        'input_ids': {1: 'sequence_length'},
        'attention_mask': {1: 'sequence_length'},
        'token_type_ids': {1: 'sequence_length'},
        'sentence_embedding': {0: 'sequence_length'},
    },
```

**Comment by kibae** [comment]
In your Python code, the shape is specified as `(1024,)`, which implies a shape of `(1024, -1)`.
However, the session information from onnxruntime-server shows the `sentence_embedding` output as `float32[-1, 1024]`. Therefore, when exporting to ONNX, you should add `'sentence_embedding': {0: 'sequence_length'}` so that onnxruntime-server's session information displays it as `float32[1024, -1]`.

**Comment by sidharthg-couture** [comment]
To make the axes [1] as dynamic, I would have to use `'sentence_embedding': {1: 'sequence_length'}` right, ottherwise it would be 
```
"outputs": {
    "sentence_embedding": "float32[-1,1024]"
  },
```
This was the output after trying the suggested `dynamic_axes` configuration.

**Comment by kibae** [comment]
@sidharthg-couture I released version 1.21.0, would you like to try again with the new version?

**Comment by Saber386** [comment]
Hey @sidharthg-couture ,Your model's output `sentence_embedding` has a fixed batch dimension (axis 0 = 1) with only` sequence_length` as dynamic — the server might be misreading the shape metadata and failing to correctly dereference the float values, falling back to nulls silently instead of throwing an error. A few things worth trying to narrow it down:

Re-export with a fully dynamic batch axis and test if the server returns proper floats — if it does, the bug is specifically in how the server handles fixed batch dimensions during output serialization.

`dynamic_axes={
    'input_ids': {0: 'batch_size', 1: 'sequence_length'},
    'attention_mask': {0: 'batch_size', 1: 'sequence_length'},
    'token_type_ids': {0: 'batch_size', 1: 'sequence_length'},
    'sentence_embedding': {0: 'batch_size'},
}`

Try a lower opset like opset_version=14 — opset 18 is recent and the server's bundled ORT version might not fully support all ops, causing a silent output failure.

The model runs fine via Python ORT but fails on the server, which strongly points to the output tensor → JSON serialization layer in the server's C++ code rather than the inference itself. The inference is likely succeeding but the float values are getting dropped when converting to JSON response.
